### PR TITLE
Fixes welding helmet and hardhat pepperspray protection

### PR DESCRIPTION
**Pull request by JackEnoff**
## About The Pull Request

Welding helmet/hardhat will now properly protect against pepperspray while its down

## Why It's Good For The Game

Fixes uhhh... apparently the guy that made the issue report got gitbanned or something?? Idk man, but it fixes that issue where welding helmets and hardhats protected against pepperspray while up and didnt while down.

## Changelog

:cl:
fix: Welding helmet and hardhat will now properly protect against pepperspray while its down
/:cl:

### CODE REVIEW

File: code/modules/clothing/head/hardhat.dm [hunk 1]
```diff
@@ -124,9 +124,9 @@
 	tint = 2
 	flags_inv = HIDEEYES | HIDEFACE | HIDESNOUT
 	flags_cover = HEADCOVERSEYES | HEADCOVERSMOUTH
-	visor_vars_to_toggle = VISOR_FLASHPROTECT | VISOR_TINT
+	visor_vars_to_toggle = VISOR_FLASHPROTECT | VISOR_TINT | PEPPERPROOF
```
Comment (MrMelbert) [inline code/modules/clothing/head/hardhat.dm:127]: You definitely can't use the same `PEPPERPROOF` flag in `visor_vars_to_toggle`, you would need to define a new `VISOR_PEPPERPROTECT`

**Review by MrMelbert — CHANGES_REQUESTED** [review]

File: code/modules/clothing/clothing.dm [hunk 2]
```diff
@@ -9,6 +9,7 @@
 	var/flash_protect = FLASH_PROTECTION_NONE
 	var/tint = 0 //Sets the item's level of visual impairment tint, normally set to the same as flash_protect
 	var/up = FALSE //but separated to allow items to protect but not impair vision, like space helmets
+	var/pepper_protect = FALSE //Toggles pepper protection when an item is ajusted up/down
```
Comment (MrMelbert) [inline code/modules/clothing/clothing.dm:12]: You also shouldn't *need* a whole var for this, when a visor toggles it uses `^=`, which literally toggles the bit on or off

**Review by MrMelbert — CHANGES_REQUESTED** [review]
Actually, looking at it again, none of the code changes should be necessary, all you must do is move `PEPPERPROOF` from `visor_flags_cover` to `flags_cover`

The bug seems to be that the flags are reversed mistakenly

**Comment by JackEnoff** [comment]
> Actually, looking at it again, none of the code changes should be necessary, all you must do is move `PEPPERPROOF` from `visor_flags_cover` to `flags_cover`
> 
> The bug seems to be that the flags are reversed mistakenly

This actually was one of the first things I tried in hopes that it fixed it, but it broke them even further, either making the welding helmet never pepperproof and the hardhat always pepperproof, or straight up making them pepperproof always. I assume `flags_cover` straight up doesnt like when its toggled and `visors_flags_cover` doesnt actually work for visors as intended.

**Comment by JackEnoff** [comment]
ok apparently applying `PEPPERPROOF` to `flags_cover` fixed it now? im genuinely going mental... I should probably not work on stuff at like 8 in the morning...

**Review by MrMelbert — APPROVED** [review]

**Pull request merged**